util/encoding/encoding.go

```diff
@@ -503,6 +511,20 @@ func DecodeIfNull(b []byte) ([]byte, bool) {
 	return b, false
 }
 
+// DecodeIfNotNull decodes a not-NULL value from the input buffer. If the input
+// buffer contains a not-NULL marker at the start of the buffer then it is
+// removed from the buffer and true is returned for the second
+// result. Otherwise, the buffer is returned unchanged and false is returned
+// for the second result. Note that the not-NULL marker is identical to the
+// empty string encoding, so do not use this routine where it is necessary to
+// distinguish not-NULL from the empty string.
+func DecodeIfNotNull(b []byte) ([]byte, bool) {
+	if len(b) >= 2 && b[0] == 0x00 && b[1] == 0x01 {
```

Comment:
fwiw `bytes.Equal` is written in asm, so `DecodeIfNull` should get this treatment too